### PR TITLE
[scantradunion] Improvements

### LINKED ISSUE #1591: [Scantrad-union] Missing manga from partner
**Did you read [the troubleshooting guide](https://github.com/manga-download/hakuneko/wiki/Troubleshooting)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
We can't see the manga from the partner page : https://scantrad-union.com/partenaires/

**To Reproduce**
For example : https://scantrad-union.com/manga/partenaire-second-life-ranker/ is missing

**HakuNeko (please complete the following information):**
 - Version 6.1.7

### PR DESCRIPTION
**Pull request by XenonR**
Closes #1591 
developed with 💖

### CODE REVIEW

File: src/web/mjs/connectors/ScantradUnion.mjs [hunk 1]
```diff
@@ -56,14 +68,17 @@ export default class ScantradUnion extends Connector {
...
-        this.requestOptions.headers.delete('x-referer');
-        return promise;
+    async _getMangaFromURI(uri) {
+        let request = new Request(new URL(uri.href), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div.projet-description > h2');
+        return new Manga(this, uri.pathname, this.cleanTitle(data[0].textContent));
+    }
+
+    cleanTitle(title) {
+        if (title.trim().startsWith('[Partenaire] ')) {
```
Comment (27136773) [inline src/web/mjs/connectors/ScantradUnion.mjs]: May use replace instead of search and slice

File: src/web/mjs/connectors/ScantradUnion.mjs [hunk 2]
```diff
@@ -8,17 +9,28 @@ export default class ScantradUnion extends Connector {
...
-                title: element.querySelector('h2.index-top3-title').innerText.trim()
-            };
-        });
+        let request = new Request(this.url + '/manga/', this.requestOptions);
+        let pages = await this.fetchDOM(request, 'main div.pagination li:nth-last-child(2) > a');
+        pages = pages[0].pathname.match(/[\w\W]+\/(\d+)\/$/)[1];
+
+        let data;
+        let mangas = [];
+        for (let page = 0; page <= pages; page++) {
```
Comment (27136773) [inline src/web/mjs/connectors/ScantradUnion.mjs:22]: Comparison is not type safe, convert `pages` from string to number

File: src/web/mjs/connectors/ScantradUnion.mjs [hunk 3]
```diff
@@ -27,11 +39,11 @@ export default class ScantradUnion extends Connector {
         return data.map(element => {
             let number = element.querySelector('span.chapter-number').innerText.trim();
             let title = element.querySelector('span.chapter-name').innerText.trim();
+            title.length ? title = ' - ' + title : title;
```
Comment (27136773) [inline src/web/mjs/connectors/ScantradUnion.mjs]: May use ternary operator for assignment
`title = title.length ? ' - ' + title : title`

**Review by 27136773 — APPROVED** [review]
Looks good

File: src/web/mjs/connectors/ScantradUnion.mjs (hunk 1 above)
Comment (XenonR) [reply src/web/mjs/connectors/ScantradUnion.mjs]: Wasn't aware `String.replace` was regex aware. Cool stuff.

**Pull request merged**